enclave/core/sgx/asmcommon.inc

```diff
@@ -28,13 +28,22 @@
     push %rax
     push %rdx
 
+    // Check if XSAVE/XRSTOR is supported
+    lea     oe_is_xsave_supported(%rip), %rax
+    movl    (%rax), %eax
+    cmpl    $0, %eax
+    jz      1f
+
     // Set the XSAVE_MASK
     mov $0xFFFFFFFF, %eax
     mov $0xFFFFFFFF, %edx
 
     // Restore initial enclave XSAVE state
     xrstor64 OE_XSAVE_INITIAL_STATE(%rip)
-
+    jmp     2f
+1:
+    fxrstor64 OE_XSAVE_INITIAL_STATE(%rip)
```

Comment:
Can you add a comment explaining why fxrstor64 is needed here, and not needed when the CPU does not support xsave.

Reply:
Added more comments to the conditional and the branch.